lib/super_digit.rb

```diff
@@ -1,15 +1,26 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - ? O(n)
+# Space Complexity - ? O(n)
+
+#Given an integer, we need to find the super digit of the integer.
+
 def super_digit(n)
-   
+  return n if n < 10
+
+  chars_array = n.to_s.chars 
+  int_array = chars_array.map(&:to_i) 
+  sum = int_array.sum 
+
+  super_digit(sum)
 end
-  
 
-# Time Complexity - ?
-# Space Complexity - ?
+
+# Time Complexity - O(n)
+# Space Complexity - O(1)
 def refined_super_digit(n, k)
-    
+ 
+  return  super_digit((n.to_s * k).to_i) 
```

Comment:
Note that the number of digits here will be O(n * k) so ....

You can do this with dynamic programming in O(n + k)

```ruby
  sum = super_digit(n)
  sum *= k
  return super_digit(sum)
```